src/app/components/badges/full-list-of-achievements/full-list-of-achievements.component.ts

```diff
@@ -0,0 +1,20 @@
+import { Component, OnInit } from '@angular/core';
+import { Achievelist } from 'src/app/models/achievelist';
+import { AhievementListService } from 'src/app/shared/services/ahievement-list.service';
+
+@Component({
+  selector: 'app-full-list-of-achievements',
+  templateUrl: './full-list-of-achievements.component.html',
+  styleUrls: ['./full-list-of-achievements.component.scss']
+})
+export class FullListOfAchievementsComponent implements OnInit {
+
+  list:Achievelist[] | undefined;
```

Comment:
what is this undefied for?

Reply:
It does not work without "undefined", I don't get, why.

Reply:
try to add a default value like 'list: Achievelist[] = []'
or set second type as null, not undefined